crates/oxc_minifier/src/compressor/fold.rs

```diff
@@ -180,4 +194,265 @@ impl<'a> Compressor<'a> {
 
         None
     }
+
+    fn try_fold_unary_operator<'b>(
+        &mut self,
+        unary_expr: &'b mut UnaryExpression<'a>,
+    ) -> Option<Expression<'a>> {
+        // TODO: I want compress children first, so we can fold expression like `- - 4`
+        // But our ast will visit more than once, is there better way?
+        self.fold_expression(&mut unary_expr.argument);
+
+        let tri_kind = get_boolean_value(&unary_expr.argument);
+
+        if tri_kind == Tri::Unknown {
+            return None;
+        }
+
+        match unary_expr.operator {
+            UnaryOperator::LogicalNot => {
+                if let Expression::NumberLiteral(number_literal) = &unary_expr.argument {
+                    let value = number_literal.value;
+                    if value == 0_f64 || (value - 1_f64).abs() < f64::EPSILON {
+                        None
+                    } else {
+                        // Tri::Unknown has been already filtered out
+                        let bool_literal =
+                            self.hir.boolean_literal(unary_expr.span, tri_kind != Tri::True);
+                        Some(self.hir.literal_boolean_expression(bool_literal))
+                    }
+                } else {
+                    None
+                }
+            }
+            UnaryOperator::UnaryPlus => match &unary_expr.argument {
+                Expression::NumberLiteral(number_literal) => {
+                    let number_literal = self.hir.number_literal(
+                        unary_expr.span,
+                        number_literal.value,
+                        number_literal.raw,
+                        number_literal.base,
+                    );
+                    Some(self.hir.literal_number_expression(number_literal))
+                }
+                Expression::Identifier(ident) => {
+                    if matches!(ident.name.as_str(), "NaN" | "Infinity") {
+                        self.try_detach_unary_op(unary_expr)
+                    } else {
+                        None
+                    }
+                }
+                _ => {
+                    if let Some(value) = get_number_value(&unary_expr.argument) {
```

Comment:
Is it good to mix convert number in the `try_fold_unary_operator ` function? 

I am not found a good way like:

```rust
Expression::UnaryExpression(unary_expr) => match unary_expr.operator {
          UnaryOperator::UnaryPlus=> {
              // maybe reduce op function need to be here 
              // and keep try_fold_typeof only fold expression.
              self.try_fold_typeof(unary_expr.span, &unary_expr.argument)
          }
}
```

[`tryReduceOperandsForOp`](https://github.com/google/closure-compiler/blob/a4c880032fba961f7a6c06ef99daa3641810bfdd/src/com/google/javascript/jscomp/PeepholeFoldConstants.java#L92)

Reply:
I'm not sure 😅